### DISCUSSION TITLE
Fetch external https API every x seconds and parse resulting JSON?

### DISCUSSION THREAD

**Discussion by andy71** [arendst/Tasmota#17829] [Q&A] [answered]
Hi, I am new to Tasmota and will get some Shelly Duo RGBW bulbs.

Will it be possible to add a script, module, rule or something like this to do the following every x seconds...
- fetch external https API URL and get a JSON result
- parse the JSON and depending on the values turn on the bulb, set color and maybe blink with specific frequency, and turn off after x minutes/seconds

It would also be great if it would be possible to add a configuration page for some URL parameters to the web ui.

If it is possible, what would be the best way to solve it? A berry script? Does anybody knows an example for this?

Thanks in advance.

**Comment by barbudor** [comment, marked as answer]
Https will be possible only on ESP32 so not on esp8266 where you will have to limit yourself to http

Berry is available also only on ESP32

A rule could be enough if the parsing is simple
Otherwise you could use Scripting-Language

**Reply by sfromis** [reply]
You could also have a gateway device (possibly ESP32 with Tasmota+Berry) to do the external communication, and then have your ESP8266-based  devices go via that gateway.
